lib/utils.js

```diff
@@ -30,22 +38,22 @@ exports.parseMongoConnectionURL = function (mongoConnectionURL) {
   //host:port host2:port2
   var hostList = hosts.split(",");
 
-  //user, password
-  var authUser = auth.split(":")[0];
-  var authPassword = auth.split(":")[1];
-
-  result.database = {};
-  result.database.driver_options = {};
-  result.database.auth = {};
-  result.database.auth.user = authUser;
-  result.database.auth.pass = authPassword;
-  result.database.auth.source = databaseName;
   result.database.name = databaseName;
 
-  assert.ok(authUser !== undefined);
-  assert.ok(authPassword !== undefined);
   assert.ok(databaseName !== undefined);
 
+  if (auth) {
+    //user, password
+    var authUser = auth.split(":")[0];
+    var authPassword = auth.split(":")[1];
+
+    result.database.auth = {
+      user: authUser,
+      pass: authPassword,
+      source: databaseName
+    };
+  }
+  
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)